src/Mutations/ImportModelsAsync.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mrap\GraphCool\Mutations;
+
+use GraphQL\Error\Error;
+use GraphQL\Type\Definition\ResolveInfo;
+use Mrap\GraphCool\DataSource\DB;
+use Mrap\GraphCool\Definition\ModelBased;
+use Mrap\GraphCool\Definition\Mutation;
+use Mrap\GraphCool\Types\Type;
+use Mrap\GraphCool\Utils\Authorization;
+use Mrap\GraphCool\Utils\JwtAuthentication;
+use RuntimeException;
+use function Mrap\GraphCool\pluralize;
+
+class ImportModelsAsync extends Mutation
+{
+    use ModelBased;
+
+    public function __construct(?string $model = null)
+    {
+        if ($model === null) {
+            throw new RuntimeException(__METHOD__.': parameter $model may not be null for ModelBased mutations.');
+        }
+        $plural = pluralize($model);
+        $this->name = 'import'.$plural.'Async';
+        $this->model = $model;
+
+        $this->config = [
+            'type' => Type::string(),
+            'description' => 'Import a list of ' . $plural . ' from a spreadsheet - in the background. Will return the job_id of the background job.',
+            'args' => $this->importArgs($model)
+        ];
+    }
+
+    public function resolve(array $rootValue, array $args, mixed $context, ResolveInfo $info): mixed
+    {
+        Authorization::authorize('import', $this->model);
+        if (!isset($args['data_base64']) || empty($args['data_base64'])) {
+            if (($args['file']['tmp_name'] ?? null) === null) {
```

Comment:
Leerstring für `$args['file']['tmp_name']` liefert hier `true`, ist das gewollt? 
Man könnte ansonsten einfach auf `empty()` prüfen.

Reply:
tut es nicht.

```$args = [
	'file' => [
		'tmp_name' => ''
	],
];
var_dump(($args['file']['tmp_name'] ?? null) === null);```

=> `boolean false`

Reply:
Aber wird als tmp_name nicht ein String erwartet? 
Da liefert mir diese Überprüfung auch false zurück.
https://onlinephp.io?s=s7EvyCjg5VJJLEovjlZPy8xJVY-NVi_JLYjPS8wFshVsFZRKUotLlKx5ucoSi-JTSnMLNHCq1kRWhVuZgr29Ql5pTo6mgq2tLYRlDQA%2C&v=8.1.12

Reply:
> tut es nicht.
> 
> ```
> $args = [
> 	'file' => [
> 		'tmp_name' => ''
> 	],
> ];
> var_dump(($args['file']['tmp_name'] ?? null) === null);
> ```
> 
> => `boolean false`

Und stimmt hatte die Überprüfung schon abgeändert, darum kam ich auf `true`.
Aber versteh sie trotzdem noch nicht.

Reply:
Hier wird ja überprüft ob tmp_name existiert, ansonsten auf null gesetzt. Und dann überprüft man wieder auf null um als Ergebniss true oder false zu erhalten. Warum nicht isset?

Reply:
Warum isset?

Es stimmt, `!isset` liefert in diesem Fall exakt dasselbe als `($x ?? null) === null`:
```
$args = [];
var_dump([
	'===' => ($args['file']['tmp_name'] ?? null) === null,
	'empty' => empty($args['file']['tmp_name']),
	'!isset' => !isset($args['file']['tmp_name']),
]);

$args = [
	'file' => [],
];
var_dump([
	'===' => ($args['file']['tmp_name'] ?? null) === null,
	'empty' => empty($args['file']['tmp_name']),
	'!isset' => !isset($args['file']['tmp_name']),
]);

$args = [
	'file' => [
		'tmp_name' => ''
	],
];
var_dump([
	'===' => ($args['file']['tmp_name'] ?? null) === null,
	'empty' => empty($args['file']['tmp_name']),
	'!isset' => !isset($args['file']['tmp_name']),
]);
```
Aber für `!isset` musste ich zuerst dieses script schreiben um mir da wirklich sicher zu sein, dass es so funktioniert - und für `===` ist es mir intuitiv sofort klar, dass es so sein muss...

Und da ich diesen Code in ein paar Monaten sicher wieder mal lesen und verstehen muss... hilft es mir, wenn ich dann net wieder isset() überprüfen muss, weil ichs wieder vergessen habe ;)

Reply:
OK :D null coalescing sonst null und dann check auf null ist mir einfach zu viel null vermutlich :D